src/engine/runtime.js

```diff
@@ -1218,29 +1222,52 @@ class Runtime extends EventEmitter {
             argJSON.check = argTypeInfo.check;
         }
 
-        const shadowType = (argInfo.menu ?
-            this._makeExtensionMenuId(argInfo.menu, context.categoryInfo.id) :
-            argTypeInfo.shadowType);
-        const fieldType = argInfo.menu || argTypeInfo.fieldType;
+        let valueName;
+        let shadowType;
+        let fieldName;
+        if (argInfo.menu) {
+            const menuInfo = argInfo.menu && context.categoryInfo.menuInfo[argInfo.menu];
+            if (menuInfo.acceptReporters) {
+                valueName = placeholder;
+                shadowType = this._makeExtensionMenuId(argInfo.menu, context.categoryInfo.id);
+                fieldName = argInfo.menu;
+            } else {
+                argJSON.type = 'field_dropdown';
+                argJSON.options = menuInfo.items;
+                valueName = null;
```

Comment:
why not just initialize these to `null` when declaring them above?

Reply:
this is a possibly-controversial habit of mine: if I have a variable that gets initialized in one of several ways I generally leave it uninitialized and then make sure to initialize it in each path. That way if something happens to the control flow (like a `break` gets added or something) and results in the possibility of skipping initialization of the variable, an IDE or linter can report that the variable might be uninitialized below. Initializing to `null` makes sure that the variable is definitely initialized but doesn't guarantee that it's initialized to a correct value...

TL;DR: it's a habit of mine that I like for maintenance reasons but if other people don't like it then I'm happy to switch.